### ISSUE TITLE
Sidebar link being generated incorrectly, No route matches

### ISSUE THREAD

**Issue by rhysdavies1994** [thoughtbot/administrate#400]
In my app, my associations are set up so each User can have many Equipments, and each Equipment belongs to one User.

When running administrate, going to the admin index page which is admin/users it comes up with the following error:
ActionController::UrlGenerationError in Admin::Equipment#index
No route matches {:action=>"show", :controller=>"admin/equipment"} missing required keys: [:id]

Same error occurs on admin/equipment route

I am able to get to admin/users/1, which shows the associated user and its equipments, although in the sidebar i notice the link for Equipment is http://localhost:3000/admin/equipment/1

This link works to show the first equipment object, but Shouldn't that be just admin/equipment?
and why isn't my admin_equipment_index action working with equipment/index it looks like its using equipment/show?

See routes below
![routes](https://cloud.githubusercontent.com/assets/7247102/12226816/e2e85420-b85c-11e5-8fc8-e61e88fdc17d.PNG)

**Comment by beydogan** [comment]
Having the same issue on a model named 'PatientData' probably its because of pluralization.

**Comment by tmecklem** [comment]
I submitted a PR for this: https://github.com/thoughtbot/administrate/pull/414

**Comment by macowie** [comment]
Just hit this with a model named "Series"

**Comment by pablo-co** [comment]
Same here, same model 'Equipment'.

**Comment by minimul** [comment]
Same as @pablo-co on model Equipment

**Comment by minimul** [comment]
Work around. 

I already was using a custom layout e.g. `rails g administrate:views:layout` so in `app/views/admin/application/_sidebar.html.erb` I made the following mod which is based off of @tmecklem pr #414 

```erb
<%#
# Sidebar

This partial is used to display the sidebar in Administrate.
By default, the sidebar contains navigation links
for all resources in the admin dashboard,
as defined by the DashboardManifest.
%>

<ul class="sidebar__list">
  <% DashboardManifest::DASHBOARDS.each do |resource| %>
    <li>
      <%= link_to(
        display_resource_name(resource),
        [namespace, resource == :equipment ? ActiveModel::Naming.route_key(resource.to_s.classify.constantize) : resource],
        class: "sidebar__link sidebar__link--#{nav_link_state(resource)}"
      ) %>
    </li>
  <% end %>
</ul>
```

The line that is changed:

```ruby
[namespace, resource == :equipment ? ActiveModel::Naming.route_key(resource.to_s.classify.constantize) : resource],
```

**Comment by tmecklem** [comment]
I haven't heard anything about PR #414. It seems like the fix is still valid and needed. @graysonwright can I get an up or down on that PR for this issue?

**Comment by brndn4** [comment]
Upgrading to 0.7.0 broke this fix for me. I got around it by modifying this workaround slightly like so:

`[namespace, resource.to_s == 'series' ? ActiveModel::Naming.route_key(resource.to_s.classify.constantize) : resource.path],`

**Comment by hcarreras** [comment]
Just hit it today. Seems like this bug is still active.

**Comment by nickcharlton** [comment]
The fix (#934) isn't actually in a release yet, as it went out just after [v0.8.1](https://github.com/thoughtbot/administrate/releases/tag/v0.8.1). It should be in soon, but in the meantime, I would recommend installing from GitHub directly.

I'm going to close this for now, but please open a new issue if it's still occurring when you do that.
